### ISSUE TITLE
About the Imitation Learning in the paper

### ISSUE THREAD

**Issue by shuoyinn** [microsoft/ToRA#18]
Hello, thank you for your contributions! ToRA is really a fantastic work and I've learned so much from it!

Well, I have a question about the term “imitation learning" in your paper. Given a problem as the source sequence and the final $\tau$ as the target sequence, is there something different between the "imitation learning" and the common seq2seq training for decoder-based LLMs? 

I guess it is purely in the output $o_i$, which is not predicted or generated by the LLMs, as a similar work MathCoder mentioned in its Section 2.2
![image](https://github.com/microsoft/ToRA/assets/39686155/e0778402-c049-419e-9f12-652f3fecef54)

However, I did not see the same description in your paper and did not find you extracting and masking the ```output``` part in your finetuning script. Could you please explain it?

**Comment by ZubinGou** [comment]
Thanks for your question!

We use imitation learning to emphasize "learning from the teacher's trajectory", and the training is the same as traditional seq2seq.

It is true that $o_i$ is not model-generated, but since this part is very short, we did not mask its loss in the implementation, and we do not expect it to have an impact on model performance, you may try it out~

**Comment by shuoyinn** [comment]
I got it. Thank you for your reply!



***@***.***
 
From: Zhibin Gou
Date: 2024-01-03 12:26
To: microsoft/ToRA
CC: Shuo Yin; Author
Subject: Re: [microsoft/ToRA] About the Imitation Learning in the paper (Issue #18)
Thanks for your question!
We use imitation learning to emphasize "learning from the teacher's trajectory", and the training is the same as traditional seq2seq.
It is true that $o_i$ is not model-generated, but since this part is very short, we did not mask its loss in the implementation, and we do not expect it to have an impact on model performance, you may try it out~
